src/java/org/apache/cassandra/tools/CMSOfflineTool.java

```diff
@@ -0,0 +1,434 @@
+                     Files.createTempFile("clustermetadata", "dump");
+
+
+            try (FileOutputStreamPlus out = new FileOutputStreamPlus(p))
+            {
+                VerboseMetadataSerializer.serialize(ClusterMetadata.serializer,
+                                                    metadata,
+                                                    out,
+                                                    getSerializationVersion());
+                output.out.println("Updated cluster metadata written to file " + p.toAbsolutePath());
+            }
+        }
+
+        Version getSerializationVersion()
+        {
+            return serializationVersion != null ? serializationVersion : NodeVersion.CURRENT.serializationVersion();
+        }
+    }
+
+    public static class ClusterMetadataToolHelp extends Help implements ClusterMetadataToolRunnable
+    {
+
+        @Override
+        public void run(Output output)
+        {
+            run();
+        }
+    }
+
+    @Command(name = "addtocms", description = "Makes a node as CMS member")
+    public static class AddToCMS extends ClusterMetadataToolCmd
+    {
+        @Option(name = { "-ip", "--ip-address" }, description = "IP address of node to make CMS", required = true)
+        private String ipAddress;
+
+        @Option(type = OptionType.COMMAND, name = { "-o", "--output-file" }, description = "Ouput file path for storing the updated Cluster Metadata")
+        private String outputFilePath;
+
+        @Override
+        public void run(Output output) throws IOException
+        {
+            ClusterMetadata metadata = parseClusterMetadata();
+            InetAddressAndPort nodeAddress = InetAddressAndPort.getByNameUnchecked(ipAddress);
+            metadata = makeCMS(metadata, nodeAddress);
+            writeMetadata(output, metadata, outputFilePath);
+        }
+
+        ClusterMetadata makeCMS(ClusterMetadata metadata, InetAddressAndPort endpoint)
+        {
+            ReplicationParams metaParams = ReplicationParams.meta(metadata);
+            DataPlacement.Builder builder = metadata.placements.get(metaParams).unbuild();
+
+            Replica newCMS = MetaStrategy.replica(endpoint);
+            builder.withReadReplica(metadata.epoch, newCMS)
+                   .withWriteReplica(metadata.epoch, newCMS);
+            return metadata.transformer().with(metadata.placements.unbuild().with(metaParams,
+                                                                                  builder.build())
+                                                                  .build())
+                           .build().metadata;
+        }
+    }
+
+    @Command(name = "assigntokens", description = "Assigns a token for given instance")
+    public static class AssignTokens extends ClusterMetadataToolCmd
+    {
+        @Option(name = { "-ip", "--ip-address" }, description = "IP address of endpoint. Port can be specified as well.", required = true)
```

Comment:
nit: describe how to specify port, i.e. with semicolon, for completenesss

Reply:
Done